File: src/main/java/io/github/tanguygab/conditionalactions/conditions/types/StringCondition.java
package io.github.tanguygab.conditionalactions.conditions.types;

import org.bukkit.OfflinePlayer;

import java.util.Map;
import java.util.function.BiFunction;

public class StringCondition extends ConditionType {

    private final BiFunction<String, String, Boolean> function;

    public StringCondition(String[] input, BiFunction<String,String,Boolean> function) {
        super(input);
        this.function = function;
    }

    @Override
    public boolean isMet(OfflinePlayer p, Map<String, String> replacements) {
        return function.apply(parseLeft(p,replacements), parseRight(p,replacements));
    }

}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/conditions/types/StringCondition.kt
package io.github.tanguygab.conditionalactions.conditions.types

import org.bukkit.OfflinePlayer

class StringCondition(input: List<String>, private val function: (String, String) -> Boolean) : ConditionType(input) {
    override fun isMet(player: OfflinePlayer?) = function(parseLeft(player), parseRight(player))
}
